torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -234,23 +318,66 @@ def do_auto_functionalize(
             normalized_kwargs[arg.name] = arg.default_value
 
     unwrapped_kwargs = ctx.unwrap_tensors(normalized_kwargs)  # type: ignore[arg-type]
+
     if "self" in unwrapped_kwargs or "self_" in unwrapped_kwargs:
         warnings.warn(
             "Using `self` or `self_` as an argument in the definition of custom ops may lead to ambiguous parsing. "
             "Please consider using a different name for this argument to avoid potential issues."
         )
+    # List of the name of args that get mutated (according to the schema)
+    mutable_args_names = get_mutable_arg_names(op)
+
+    all_basis = set()
+    all_basis_addresses: set[int] = set()
+    basis: Dict[str, Any] = {}
+
+    for arg_name in mutable_args_names:
+        arg = normalized_kwargs[arg_name]
+        if arg is None:
+            continue
+
+        if isinstance(arg, list):
+            basis[arg_name] = []
+            for tensor in arg:
+                if tensor is None:
+                    basis[arg_name].append(None)
+                    continue
+
+                base = tensor if tensor._base is None else tensor._base
```

Comment:
We should check if tensor._base is available under torch.inference_mode(). If it's not then... we're gonna need to add more infra to keep track of bases.

Reply:
with torch.inference_mode():
    x = torch.randn(3)
    f(x)